src/main/java/org/folio/service/budget/BudgetService.java

```diff
@@ -60,146 +59,109 @@ public void deleteById(String id, Context vertxContext, Map<String, String> head
-          logger.info("unlinkGroupFundFiscalYears:: Group fund fiscal years have been successfully unlinked by budget id {}", id);
-          promise.complete();
-        }
-      });
-    return promise.future();
+    return conn.execute(sql, Tuple.of(UUID.fromString(id)))
+      .onSuccess(ecList -> logger.info("unlinkGroupFundFiscalYears:: Group fund fiscal years have been successfully unlinked by budget id {}", id))
+      .onFailure(e -> logger.error("unlinkGroupFundFiscalYears:: Failed to update group_fund_fiscal_year by budget id {}", id, e))
+      .mapEmpty();
   }
 
-  public Future<Budget> getBudgetByFundIdAndFiscalYearId(String fiscalYearId, String fundId, DBClient dbClient, boolean withTx) {
+  public Future<Budget> getBudgetByFundIdAndFiscalYearId(String fiscalYearId, String fundId, DBConn conn) {
     logger.debug("getBudgetByFundIdAndFiscalYearId:: Trying to get budget by fund id {} and fiscal year id {}", fundId, fiscalYearId);
-    Promise<Budget> promise = Promise.promise();
 
     Criterion criterion = new CriterionBuilder().with("fundId", fundId)
       .with("fiscalYearId", fiscalYearId)
       .build();
-    Future.succeededFuture()
-      .compose(v -> {
-        if (withTx) {
-          return budgetDAO.getBudgetsTx(criterion, dbClient);
-        }
-        return budgetDAO.getBudgets(criterion, dbClient);
-      })
-      .onComplete(reply -> {
-        if (reply.failed()) {
-          logger.error("getBudgetByFundIdAndFiscalYearId:: Getting budget by fund id {} and fiscal year id {} failed", fundId, fiscalYearId, reply.cause());
-          handleFailure(promise, reply);
-        } else if (reply.result().isEmpty()) {
+    return budgetDAO.getBudgets(criterion, conn)
+      .map(budgets -> {
+        if (budgets.isEmpty()) {
           logger.error("getBudgetByFundIdAndFiscalYearId:: Budget not found for fundId {} and fiscalYearId {}", fundId, fiscalYearId);
-          promise.fail(new HttpException(Response.Status.NOT_FOUND.getStatusCode(),
-              JsonObject.mapFrom(BUDGET_NOT_FOUND_FOR_TRANSACTION.toError()).encodePrettily()));
+          throw new HttpException(Response.Status.NOT_FOUND.getStatusCode(),
+            JsonObject.mapFrom(BUDGET_NOT_FOUND_FOR_TRANSACTION.toError()).encodePrettily());
         } else {
           logger.info("getBudgetByFundIdAndFiscalYearId:: Successfully retrieved budget by fund id {} and fiscal year id {}", fundId, fiscalYearId);
-          promise.complete(reply.result().get(0));
+          return budgets.get(0);
         }
-      });
-
-    return promise.future();
+      })
+      .onFailure(e -> logger.error("getBudgetByFundIdAndFiscalYearId:: Getting budget by fund id {} and fiscal year id {} failed", fundId, fiscalYearId, e));
   }
 
-  public Future<Budget> getBudgetByFiscalYearIdAndFundIdForUpdate(String fiscalYearId, String fundId, DBClient dbClient) {
+  public Future<Budget> getBudgetByFiscalYearIdAndFundIdForUpdate(String fiscalYearId, String fundId, DBConn conn) {
```

Comment:
maybe it makes sense to add java doc and specify that this method should be executed only in transactional context, because it has SELECT FOR UPDATE that requires an transaction opened.